presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -514,6 +513,48 @@ private static Slice formatDatetime(ISOChronology chronology, Locale locale, lon
         }
     }
 
+    @Description("Returns the current Unix timestamp in seconds (seconds since Unix epoch)")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long unixTimestamp(ConnectorSession session)
+    {
+        return MILLISECONDS.toSeconds(session.getStartTime());
```

Comment:
That would mean that `current_date` is flawed in the same way you suggest.

More importantly, I run the test you proposed and the results are OK:

```
presto> select unix_timestamp();
   _col0    
------------
 1462523739 
(1 row)

Query 20160506_083539_00000_w79kp, FINISHED, 1 node
Splits: 1 total, 1 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

presto> select unix_timestamp();
   _col0    
------------
 1462523742 
(1 row)

Query 20160506_083542_00001_w79kp, FINISHED, 1 node
Splits: 1 total, 0 done (0.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]
```